src/main/java/io/cryostat/net/TargetConnectionManager.java

```diff
@@ -89,42 +88,7 @@
                         .executor(executor)
                         .scheduler(scheduler)
                         .expireAfterAccess(ttl)
-                        .removalListener(
-                                new RemovalListener<ConnectionDescriptor, JFRConnection>() {
-                                    @Override
-                                    public void onRemoval(
-                                            ConnectionDescriptor descriptor,
-                                            JFRConnection connection,
-                                            RemovalCause cause) {
-                                        if (descriptor == null) {
-                                            logger.warn(
-                                                    "Connection eviction triggered with null descriptor");
-                                            return;
-                                        }
-                                        if (connection == null) {
-                                            logger.warn(
-                                                    "Connection eviction triggered with null connection");
-                                            return;
-                                        }
```

Comment:
What's the reason behind not also including these two `null` checks in the new removal listener, `this::closeConnection`? Is it because the warning is useless since an `NPE` would be triggered if either the `descriptor` or `connection` were actually `null`?

Reply:
In practice I don't think the Caffeine cache will ever actually end up calling this removal listener with either of those null. But, I also don't have a good reason to remove the guard - the log message (should probably be an Error level) and early return are safer, albeit a little paranoid.